discovery/service.go

```diff
@@ -714,11 +721,18 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 		// announcement. If we didn't receive the opposite half of the
 		// proof than we should store it this one, and wait for
 		// opposite to be received.
-		oppositeKey := newProofKey(chanInfo.ChannelID, !nMsg.isRemote)
-		oppositeProof, ok := d.waitingProofs[oppositeKey]
+		proof := newWaitingProof(nMsg.isRemote, msg)
+		oppositeKey := proof.oppositeKey()
+		oppositeProofIndex, ok := d.proofToIndex[oppositeKey]
 		if !ok {
-			key := newProofKey(chanInfo.ChannelID, nMsg.isRemote)
-			d.waitingProofs[key] = msg
+			index, err := d.proofStore.Add(proof)
+			if err != nil {
+				nMsg.err <- errors.Errorf("unable to store "+
+					"the proof for short_chan_id=%v: %v",
+					shortChanID, err)
+				return nil
```

Comment:
Can you elaborate please? As far as I can tell, error is returned 3 lines above.